cpu/atmega328p/periph/adc.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Laurent Navet <laurent.navet@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_atmega328p
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation for atmega328p mcu
+ *
+ * @author      Laurent Navet <laurent.navet@gmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+
+int adc_init(adc_t line)
+{
+
+    /* check if the line is valid */
+    if (line >= ADC_NUMOF) {
+        return -1;
+    }
+
+    // Is it needed ...?
+//    DDRD = 0x00;  // configure PORTA as input
+//    DIDR0 = 0x00;           //Digital input disabled on all ADC ports
```

Comment:
- Improved adc by setting DDRD & DIDR0 as pointed by @plushvoxel 
- replaced REG |= _BV(REGX) avr style by REG |= (1 << REGX) notation which is widely used in RIOT.
- added analogRead() arduino sketches support in sys/arduino